x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -537,10 +537,31 @@ public Literal visitIntervalLiteral(IntervalLiteralContext ctx) {
 
         DataType intervalType = Intervals.intervalType(source(interval), leading, trailing);
 
-        boolean negative = interval.sign != null && interval.sign.getType() == SqlBaseParser.MINUS;
+        // negation outside the interval - use xor
+        boolean negative = false;
+
+        ParserRuleContext parentCtx = interval.getParent();
+        if (parentCtx != null) {
```

Comment:
What is this nested `if` block doing?

Reply:
Because of the structure of the grammar it's a way to get the `-` symbol from the immediate parent context in the hierarchy that could possible contain it.